source/Applications/Advanced/MaskPointCloud/MaskPointCloud.cpp

```diff
@@ -0,0 +1,221 @@
+    {
+        pcl::PointCloud<pcl::PointXYZRGB>::Ptr cloudPTR(new pcl::PointCloud<pcl::PointXYZRGB>);
+        *cloudPTR = pointCloud;
+        std::cout << "Running PCL visualizer. Block until window closes" << std::endl;
+        pcl::visualization::CloudViewer viewer("Simple Cloud Viewer");
+        viewer.showCloud(cloudPTR);
+        std::cout << "Press r to centre and zoom the viewer so that the entire cloud is visible" << std::endl;
+        std::cout << "Press q to me exit the viewer application" << std::endl;
+        while(!viewer.wasStopped())
+        {
+        }
+    }
+
+    cv::Mat pointCloudToCvZ(const pcl::PointCloud<pcl::PointXYZRGB> &pointCloud)
+    {
+        // Getting min and max values for X, Y, Z images
+        float zMax = -1;
+        float zMin = 1000000;
+        for(size_t i = 0; i < pointCloud.height; i++)
+        {
+            for(size_t j = 0; j < pointCloud.width; j++)
+            {
+                if(pointCloud(j, i).z > zMax)
+                {
+                    zMax = pointCloud(j, i).z;
+                }
+                if(pointCloud(j, i).z < zMin)
+                {
+                    zMin = pointCloud(j, i).z;
+                }
+            }
+        }
+
+        // Filling in OpenCV matrix with the cloud data
+        cv::Mat z(pointCloud.height, pointCloud.width, CV_8UC1, cv::Scalar(0)); // NOLINT(hicpp-signed-bitwise)
+        for(size_t i = 0; i < pointCloud.height; i++)
+        {
+            for(size_t j = 0; j < pointCloud.width; j++)
+            {
+                if(std::isnan(pointCloud(j, i).z))
+                {
+                    z.at<uchar>(i, j) = 0;
+                }
+                else
+                {
+                    z.at<uchar>(i, j) =
+                        static_cast<unsigned char>((255.0F * (pointCloud(j, i).z - zMin) / (zMax - zMin)));
+                }
+            }
+        }
+
+        // Applying color map
+        cv::Mat zColorMap;
+        cv::applyColorMap(z, zColorMap, cv::COLORMAP_VIRIDIS);
+
+        // Setting invalid points (nan) to black
+        for(size_t i = 0; i < pointCloud.height; i++)
+        {
+            for(size_t j = 0; j < pointCloud.width; j++)
+            {
+                if(std::isnan(pointCloud(j, i).z))
+                {
+                    auto &zRGB = zColorMap.at<cv::Vec3b>(i, j);
+                    zRGB[0] = 0;
+                    zRGB[1] = 0;
+                    zRGB[2] = 0;
+                }
+            }
+        }
+        return zColorMap;
+    }
+
+    pcl::PointCloud<pcl::PointXYZRGB> maskPointCloud(const Zivid::PointCloud &pointCloud, const cv::Mat &mask)
+    {
+        const auto data = pointCloud.copyPointsXYZColorsRGBA();
+        const int height = data.height();
+        const int width = data.width();
+
+        // Creating point cloud structure
+        pcl::PointCloud<pcl::PointXYZRGB> maskedPointCloud(width, height);
+        maskedPointCloud.is_dense = false;
```

Comment:
Is there a rule of thumb when we use sparse point clouds and when we do otherwise?

Reply:
From pcl doc: 
bool  is_dense
-- | --
True if no points are invalid (e.g., have NaN or Inf values).